From: Helge Bahmann <hcb@chaoticmind.net>
Subject: Repair and reactivate why3ide

diff -u frama-clang-0.0.7/reorder_defs.ml frama-clang-0.0.7x/reorder_defs.ml
--- frama-clang-0.0.7/reorder_defs.ml	2019-08-30 09:15:53.000000000 +0200
+++ frama-clang-0.0.7x/reorder_defs.ml	2019-11-17 19:43:37.805400073 +0100
@@ -307,7 +307,7 @@
 *)

 class compute_deps =
-  let open Cil in
+(*   let open Cil in *)
   let is_global s = Extlib.string_prefix ~strict:true "_Z" s in
   object(self)
     inherit Cabsvisit.nopCabsVisitor
Gemeinsame Unterverzeichnisse: frama-clang-0.0.7/share und frama-clang-0.0.7x/share.
Nur in frama-clang-0.0.7x: top.
